src/main/java/seedu/address/storage/JsonAdaptedStudent.java

```diff
@@ -0,0 +1,158 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.id.StudentId;
+import seedu.address.model.profbook.Address;
+import seedu.address.model.profbook.Email;
+import seedu.address.model.profbook.Name;
+import seedu.address.model.profbook.Phone;
+import seedu.address.model.profbook.Student;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.taskmanager.Deadline;
+import seedu.address.model.taskmanager.Task;
+import seedu.address.model.taskmanager.TaskList;
+import seedu.address.model.taskmanager.ToDo;
+
+
+/**
+ * Jackson-friendly version of {@link Student}.
+ */
+public class JsonAdaptedStudent {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Student's %s field is missing!";
+
+    private final String name;
+    private final String email;
+    private final String phone;
+
+    // Data fields
+    private final String address;
+    private final Set<JsonAdaptedTag> tags = new HashSet<>();
+    private final String id;
+
+    private final Set<JsonAdaptedTasks> tasks = new HashSet<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedStudent} with the given person details.
+     */
+    @JsonCreator
+    public JsonAdaptedStudent(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
+                             @JsonProperty("email") String email, @JsonProperty("address") String address,
+                              @JsonProperty("id") String id, @JsonProperty("tags") List<JsonAdaptedTag> tags,
+                              @JsonProperty("tasks") List<JsonAdaptedTasks> tasks) {
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.address = address;
+        this.id = id;
+        if (tags != null) {
```

Comment:
I believe our students shouldnt store tags?

Reply:
This will be removed soon